addons/mrp/tests/test_byproduct.py

```diff
@@ -407,3 +420,65 @@ def test_check_byproducts_cost_share_02(self):
         mo.move_byproduct_ids[0].quantity_done = 1
         mo.button_mark_done()
         self.assertEqual(mo.state, 'done')
+
+    def test_01_check_byproducts_update(self):
+        """
+        Test that check byproducts update in stock move should also reflect in stock move line(Product moves).
+        Case 1: Update Product From Non Tracked Product to Non Tracked Product.
+        """
+        # Create new MO
+        mo_form = Form(self.env['mrp.production'])
+        mo_form.product_id = self.product_a
+        mo_form.product_qty = 1.0
+        mo = mo_form.save()
+        mo.action_confirm()
+
+        mo.move_byproduct_ids.write({'product_id': self.product_c_id})
+        mo.button_mark_done()
+        self.assertEqual(mo.move_byproduct_ids.product_id, mo.move_byproduct_ids.move_line_ids.product_id)
+
+    def test_02_check_byproducts_update(self):
+        """
+        Case 2: Update Product From Tracked Product to Non Tracked Product.
+        """
+        self.bom_byproduct.byproduct_ids[0].product_id = self.produced_serial.id
+        self.bom_byproduct.byproduct_ids[0].product_qty = 2
+        mo = self.env["mrp.production"].create({
+            'product_id': self.product_a.id,
+            'product_qty': 1.0,
+            'bom_id': self.bom_byproduct.id,
+        })
+        mo_form = Form(mo)
```

Comment:
useless in this situation

Reply:
Hello @Whenrow , are you talking about `def test_02_check_byproducts_update(self)` ?
Should I remove this test case?

Let me know if I misunderstood.

Thanks!